platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultVariant.java

```diff
@@ -49,17 +49,19 @@ public class DefaultVariant implements ConfigurationVariantInternal {
     private Factory<List<PublishArtifact>> lazyArtifacts;
     @Nullable private String description;
 
-    public DefaultVariant(Describable parentDisplayName,
-                          String name,
-                          AttributeContainerInternal parentAttributes,
-                          NotationParser<Object, ConfigurablePublishArtifact> artifactNotationParser,
-                          FileCollectionFactory fileCollectionFactory,
-                          AttributesFactory cache,
-                          DomainObjectCollectionFactory domainObjectCollectionFactory,
-                          TaskDependencyFactory taskDependencyFactory) {
+    public DefaultVariant(
+        Describable parentDisplayName,
+        String name,
+        AttributeContainerInternal parentAttributes,
+        NotationParser<Object, ConfigurablePublishArtifact> artifactNotationParser,
+        FileCollectionFactory fileCollectionFactory,
+        AttributesFactory cache,
+        DomainObjectCollectionFactory domainObjectCollectionFactory,
+        TaskDependencyFactory taskDependencyFactory
+    ) {
         this.parentDisplayName = parentDisplayName;
         this.name = name;
-        this.attributes = new FreezableAttributeContainer(cache.mutable(parentAttributes), parentDisplayName);
+        this.attributes = new FreezableAttributeContainer(cache.mutable(parentAttributes), getDisplayName());
```

Comment:
❗ This is now calling a non-static member during construction.  That call only forwards to a static call, but still, it would be safer to avoid doing this.  Why not just add a new `displayName` field? 

It also looks odd to be passing `parentAtributes` and not passing `parentDisplayName`, but I see how this change makes the error messages more accurate.  Perhaps a comment would help clarify this.

Reply:
Added a field for `displayName`